### DISCUSSION TITLE
Keep float decimal point when output is JSON

### DISCUSSION THREAD

**Discussion by V02460** [mikefarah/yq#2204] [Q&A]
How can I keep the decimal point for floats when converting YAML to JSON? The default behavior converts them to integers, which is not what I want.

Example in which the decimal point is removed:
```console
$ echo "3.0" | yq -o json '.'
3
```

**Comment by cliff-wakefield-superloop** [comment]
Did you ever find an answer for this?

**Comment by mikefarah** [comment]
Best way would be to convert the decimal to a string...

e.g. this will convert all floats to strings:

```
echo "3.0" | ./yq  '(.. | select(tag == "!!float")) tag="!!str"' -oj
```

Note that jq does the same thing with 3.0:

```
echo '{"x": 3.0}' | jq
{
  "x": 3
}
# as a string
echo '{"x": "3.0"}' | jq
{
  "x": "3.0"
}

```
